src/metabase/api/search.clj

```diff
@@ -477,26 +448,39 @@
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
 (mu/defn ^:private search-context :- SearchContext
-  [search-string   :- [:maybe ms/NonBlankString]
-   archived-string :- [:maybe ms/BooleanString]
-   table-db-id     :- [:maybe ms/PositiveInt]
-   models          :- [:maybe [:or SearchableModel [:sequential SearchableModel]]]
-   limit           :- [:maybe ms/PositiveInt]
-   offset          :- [:maybe ms/IntGreaterThanOrEqualToZero]]
+  [{:keys [search-string
+           archived-string
+           created-by
+           table-db-id
+           models
+           limit
+           offset]} :- [:map {:closed true}
+                        [:search-string                    [:maybe ms/NonBlankString]]
+                        [:archived-string {:optional true} [:maybe ms/BooleanString]]
+                        [:created-by      {:optional true} [:maybe ms/PositiveInt]]
+                        [:table-db-id     {:optional true} [:maybe ms/PositiveInt]]
+                        [:models          {:optional true} [:maybe [:or SearchableModel [:sequential SearchableModel]]]]
+                        [:limit           {:optional true} [:maybe ms/Int]]
+                        [:offset          {:optional true} [:maybe ms/Int]]]]
   (cond-> {:search-string      search-string
-           :archived?          (Boolean/parseBoolean archived-string)
-           :current-user-perms @api/*current-user-permissions-set*}
-    (some? table-db-id) (assoc :table-db-id table-db-id)
-    (some? models)      (assoc :models
-                               (apply hash-set (if (vector? models) models [models])))
-    (some? limit)       (assoc :limit-int limit)
-    (some? offset)      (assoc :offset-int offset)))
+           :current-user-perms @api/*current-user-permissions-set*
+           :archived?          (Boolean/parseBoolean archived-string)}
+    (some? created-by)      (assoc :created-by created-by)
+    (some? table-db-id)     (assoc :table-db-id table-db-id)
+    (some? models)          (assoc :models
+                                   (if models
+                                     (apply hash-set (if (vector? models) models [models]))
+                                     search-config/all-models))
+    (some? limit)           (assoc :limit-int limit)
+    (some? offset)          (assoc :offset-int offset)))
```

Comment:
Can you restore the original indentation and order of `:current-user-perms` and `:archived?`, just thinking we should keep git history simpler when we can